### PR TITLE
mgmtfn/k8splugin: add network setup unit test

### PR DESCRIPTION
**Pull request by unclejack**
This PR adds a unit test for some of the network setup functions from driver.go.

This increases the coverage from 25% to 33.4%.

### CODE REVIEW

**Comment by unclejack** [comment]
ping @abhinandanpb @erikh @jojimt @gaurav-dalvi

**Comment by erikh** [comment]
Any chance I could convince you to use gocheck for this? We are using it in a lot of the newer test systems.

File: mgmtfn/k8splugin/driver_test.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+		globalNS.Close()
+	}()
+
+	newNS, err := netns.New()
+	if err != nil {
+		t.Fatal("failed to create new network namespace")
+	}
+	defer newNS.Close()
+
+	cmd := exec.Command("sleep", "9999")
```
Comment (erikh) [inline mgmtfn/k8splugin/driver_test.go]: busybox sleep doesn't support it, but both centos and ubuntu support `sleep infinity` which is waht I'm guessing you really want here.

File: mgmtfn/k8splugin/driver_test.go [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+	// ensure the interface's name has been changed
+	newLink, err := netlink.LinkByName(newName)
+	if err != nil {
+		t.Fatalf("failed to get interface by name: %v", err)
+	}
+	defer netlink.LinkDel(newLink)
+
+	// ensure that the interface's IP address has been set properly
+	addresses, err := netlink.AddrList(newLink, netlink.FAMILY_V4)
+	ifAddr := fmt.Sprintf("%v", addresses[0].IPNet)
```
Comment (erikh) [inline mgmtfn/k8splugin/driver_test.go]: couldn't these `%v` prints just be `.String` calls? I presume they implement `fmt.Stringer`

File: mgmtfn/k8splugin/driver_test.go [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+	// ensure the default gateway route has been set properly
+	routes, err := netlink.RouteList(newLink, netlink.FAMILY_V4)
+	if err != nil {
+		t.Errorf("failed to fetch routes")
+	}
+
+	foundDefaultGW := false
+	foundStaticRoute := false
+	for _, route := range routes {
+		gw := fmt.Sprintf("%v", route.Gw)
```
Comment (erikh) [inline mgmtfn/k8splugin/driver_test.go]: same here

**Comment by erikh** [comment]
another advantage of gocheck; if we used it the test fixtures we could do the setns setup/reap so they were re-usable by other tests.

**Comment by unclejack** [comment]
@erikh: I'll make the changes as you've proposed. There's no need to convince me. The reason why I've written it like that was to make the changes as small as possible.

File: mgmtfn/k8splugin/driver_test.go [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+func TestNetSetup(t *testing.T) {
+	la := netlink.NewLinkAttrs()
+	la.Name = "testlinkfoo"
+	newName := "testlinknewname"
+	address := "192.168.68.68/24"
+	defGW := "192.168.68.1"
+	staticRoute := "192.168.32.0/24"
+
+	runtime.LockOSThread()
```
Comment (jojimt) [inline mgmtfn/k8splugin/driver_test.go]: @unclejack Just for my understanding, why do we need to lock the runtime here?

Comment (unclejack) [inline mgmtfn/k8splugin/driver_test.go]: @jojimt: This is needed to set up the new network namespace and run the process in it (lines 54 - 66).

Comment (jojimt) [inline mgmtfn/k8splugin/driver_test.go]: makes sense, thanks!

**Comment by jojimt** [comment]
Will merge once system tests pass. Thanks for taking care of this.

**Comment by erikh** [comment]
needs a kick, this was last committed when the CI was down.

**Comment by unclejack** [comment]
I'll merge this now. Joji has said he'd merge once the system tests pass. I need it for my next PR.

**Pull request merged**